File: LeetCode Solutions/Problems/524. Longest Word in Dictionary through Deleting.cpp
class Solution {
public:
    string findLongestWord(string s, vector<string>& dictionary) {
        string longestString = "";
        for(string word : dictionary){
            int wordLength = word.length(), stringLength = longestString.length();
            if(wordLength < stringLength || (wordLength == stringLength && word > longestString)){
                continue;
            }
            int position = -1;
            for(int i=0; i<wordLength; i++){
                position = s.find(word[i], position + 1);
                if(position == -1){
                    break;
                }
            }
            if(position != -1){
                longestString = word;
            }
        }
        return longestString; 
    }
};

File: LeetCode Solutions/Problems/524. Longest Word in Dictionary through Deleting.py
class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        longestString = ""
        for word in dictionary:
            wordLength, stringLength = len(word), len(longestString)
            if(wordLength < stringLength or (wordLength == stringLength and word > longestString)):
                continue
            position = -1
            for char in word:
                position = s.find(char, position + 1)
                if(position == -1):
                    break
            if(position != -1):
                longestString = word
        return longestString